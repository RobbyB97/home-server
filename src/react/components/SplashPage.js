import React from 'react'
import {NavLink} from 'react-router-dom'
import { connect } from 'react-redux'

import {splashOff} from '../actions/home'

const SplashPage = ({splash, dispatch}) => (
    <div id="splash-page" className={splash ? "":"hidden"}>
        <div class="splash-content">
            <h2 className="text-one">
                Designer. <br/> &nbsp; Developer. <br/> &nbsp;&nbsp; Problem Solver.
            </h2>
            <h3 className="text-two">
                Robby Bergers
            </h3>
            <button className="to-site" onClick={() => {
                dispatch(splashOff())
            }}>
                Splash Page button
            </button>
        </div>
    </div>
)

const mapStateToProps = (state) => {
    return {
        splash: state.home.splash
    }
  }

export default connect(mapStateToProps)(SplashPage)