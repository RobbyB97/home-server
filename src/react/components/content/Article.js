/*
   Standard Article
*/

import React from 'react'
import {Link, NavLink} from 'react-router-dom'
import {connect} from 'react-redux'

export const Article = (props) => (
    <div className="article">
        {props.title && 
            <h2 className="article__title">
                {props.title}
            </h2>
        }

        {props.text &&
            <p className="article__text">
                {props.text}
            </p>
        }

    </div>
)

export default connect()(Article)