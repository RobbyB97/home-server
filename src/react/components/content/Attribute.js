import React from 'react'
import {Link, NavLink} from 'react-router-dom'
import {connect} from 'react-redux'


const Attribute = (props) => (
    <div className="attribute">
        {
            props.image &&
                <div className="attribute__image">
                    <img src={props.image}/>
                </div>
        }
        {
            props.title && 
                <h2 className="attribute__title">
                    {props.title}
                </h2>
        }

        {   props.text &&
                <p className="attribute__text">
                    {props.text}
                </p>
        }

    </div>
)

export default Attribute