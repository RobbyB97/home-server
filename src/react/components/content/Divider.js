import React from 'react'

export class Divider extends React.Component {

    render() {
        return (
            <div className="divider">

            </div>
        )
    }
}


export default Divider