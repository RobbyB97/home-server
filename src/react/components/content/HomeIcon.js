import React from 'react'


export default class HomeIcon extends React.Component {

    render() {
        return (
            <div className="homeIcon">
                <a className="homeIcon__link" title={this.props.name} href={this.props.link}>
                    <img className="homeIcon__icon" src={this.props.icon} />
                </a>
            </div>
        )
    }
}