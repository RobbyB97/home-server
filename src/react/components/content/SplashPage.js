import React from 'react'
import {NavLink} from 'react-router-dom'
import { connect } from 'react-redux'

import {splashOff} from '../../actions/home'

export class SplashPage extends React.Component {

    componentDidMount = () => {
        setTimeout(() => {
            const text = document.querySelector('.splash__words.one')
            text.style.opacity = '1'
        }, 200)
        
        setTimeout(() => {
            const text = document.querySelector('.splash__words.two')
            text.style.opacity = '1'
        }, 800)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.three')
            text.style.opacity = '1'
        }, 1100)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.four')
            text.style.opacity = '1'
        }, 1400)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.five')
            text.style.opacity = '1'
        }, 1700)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.six')
            text.style.opacity = '1'
        }, 2000)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.seven')
            text.style.opacity = '1'
        }, 2300)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.eight')
            text.style.opacity = '1'
        }, 2900)

        setTimeout(() => {
            const text = document.querySelector('.splash__name')
            text.style.opacity = '1'
        }, 3000)

        setTimeout(() => {
            const text = document.querySelector('.splash__to-site')
            text.style.opacity = '1'
        }, 3100)

    }

    render() {
        return (
            <div id="splash" className={this.props.splash ? "":"splash__hidden"}>
                <div className="splash__background"></div>
                <div className="splash__content">
                    <p className="splash__text">
                        <span className="splash__words one">
                            What is a perfect website?&nbsp;
                        </span>
                        <span className="splash__words two">
                            The answer is a&nbsp;
                        </span>
                        <span className="splash__words three">
                            <i>memorable</i>,&nbsp;
                        </span>
                        <span className="splash__words four">
                            <i>simplistic</i>,&nbsp;
                        </span>
                        <span className="splash__words five">
                            <i>intuitive</i>,&nbsp;
                        </span>
                        <span className="splash__words six">
                            and <i>fast</i>&nbsp;
                        </span>
                        <span className="splash__words seven">
                            user experience.&nbsp;
                        </span>
                        <span className="splash__words eight">
                            My goal is to create that perfect website for your business.
                        </span>

                    </p>
        
                    <p className="splash__name">
                        Robby Bergers.
                    </p>
                    <button className="splash__to-site" onClick={() => {
                        this.props.dispatch(splashOff())
                    }}>
                        Learn more â†’
                    </button>
                </div>
            </div> 
        )
    }
}

/*
<p className="splash__text one">
                        Designer.
                    </p>
        
                    <p className="splash__text two">
                        Developer.
                    </p>
        
                    <p className="splash__text three">
                        Problem solver.
                    </p>


setTimeout(() => {
            let text = document.querySelector('.splash__text.one')
            text.style.opacity = '1'
        }, 200)

        setTimeout(() => {
            let text = document.querySelector('.splash__text.two')
            text.style.opacity = '1'
        }, 350)

        setTimeout(() => {
            let text = document.querySelector('.splash__text.three')
            text.style.opacity = '1'
        }, 500)

        setTimeout(() => {
            let text = document.querySelector('.splash__name')
            text.style.opacity = '1'
        }, 750)

        setTimeout(() => {
            let text = document.querySelector('.splash__to-site')
            text.style.opacity = '1'
        }, 850)
*/

const mapStateToProps = (state) => {
    return {
        splash: state.home.splash
    }
  }

export default connect(mapStateToProps)(SplashPage)