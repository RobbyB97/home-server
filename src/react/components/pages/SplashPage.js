import React from 'react'
import {NavLink} from 'react-router-dom'
import { connect } from 'react-redux'

import {splashOff} from '../../actions/home'

export class SplashPage extends React.Component {
    
    componentDidMount = () => {
        this.splashFadeIn()
    }

    splashOff = () => {
        this.props.splashOff()
    }

    splashFadeIn = () => {
        setTimeout(() => {
            const background = document.querySelector('.splash__background.two')
            background.style.opacity = '1'
        }, 2900)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.one')
            text.style.opacity = '1'
        }, 200)
        
        setTimeout(() => {
            const text = document.querySelector('.splash__words.two')
            text.style.opacity = '1'
        }, 800)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.three')
            text.style.opacity = '1'
        }, 1100)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.four')
            text.style.opacity = '1'
        }, 1400)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.five')
            text.style.opacity = '1'
        }, 1700)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.six')
            text.style.opacity = '1'
        }, 2000)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.seven')
            text.style.opacity = '1'
        }, 2300)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.eight')
            text.style.opacity = '1'
        }, 2900)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.nine')
            text.style.opacity = '1'
        }, 3500)

        setTimeout(() => {
            let text = document.querySelector('.splash__name')
            text.style.opacity = '1'
            text = document.querySelector('.splash__to-site')
            text.style.opacity = '1'
        }, 3900)
    }

    render() {
        return (
            <div id="splash" className={this.props.splash ? "":"splash__hidden"}>
                <div className="splash__background one"></div>
                <div className="splash__background two"></div>

                <div className="splash__content">
                    <p className="splash__text one">
                        <span className="splash__words one">
                            What is a perfect website?&nbsp;
                        </span>
                        <span className="splash__words two">
                            The answer is a&nbsp;
                        </span>
                        <span className="splash__words three">
                            <i>memorable</i>,&nbsp;
                        </span>
                        <span className="splash__words four">
                            <i>simplistic</i>,&nbsp;
                        </span>
                        <span className="splash__words five">
                            <i>intuitive</i>,&nbsp;
                        </span>
                        <span className="splash__words six">
                            and <i>fast</i>&nbsp;
                        </span>
                        <span className="splash__words seven">
                            user experience.&nbsp;
                        </span>
                        <span className="splash__words eight">
                            My goal is to create that perfect website for your business.
                        </span>
                    </p>

                    <p className="splash__text two">
                        <span className="splash__words nine">
                            Hi, my name's Robby. I'm a freelance front end developer. If you've been searching for someone to design an elegant website or web app that your clients will love, your search is over.
                        </span>
                    </p>
        
                    <p className="splash__name">
                        Robby Bergers.
                    </p>
                    <button className="splash__to-site" onClick={() => {
                        this.splashOff()
                    }}>
                        Learn more â†’
                    </button>
                </div>
            </div> 
        )
    }
}


const mapStateToProps = (state) => {
    return {
        splash: state.home.splash
    }
}

const mapDispatchToProps = (dispatch) => ({
    splashOff: () => dispatch(splashOff())
})


export default connect(mapStateToProps, mapDispatchToProps)(SplashPage)