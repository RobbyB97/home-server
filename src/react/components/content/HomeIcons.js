import React from 'react'

import HomeIcon from './HomeIcon'


export default class HomeIcons extends React.Component {

    render() {
        return (
            <div className="homeIcons">
                {this.props.icons &&
                    this.props.icons.map((icon) => {
                        <HomeIcon {...icon} />
                    })
                }
            </div>
        )
    }
}