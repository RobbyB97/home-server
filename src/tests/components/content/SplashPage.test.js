import React from 'react'
import {shallow} from 'enzyme'

import {SplashPage} from '../../../react/components/content/SplashPage'


test('Render SplashPage', () => {
    const wrapper = shallow(<SplashPage />)

    expect(wrapper).toMatchSnapshot()
})