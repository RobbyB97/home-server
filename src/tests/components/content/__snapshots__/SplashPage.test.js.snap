// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SplashPage 1`] = `
<div
  className="splash__hidden"
  id="splash"
>
  <div
    className="splash__background"
  />
  <div
    className="splash__content"
  >
    <p
      className="splash__text"
    >
      <span
        className="splash__words one"
      >
        What is a perfect website? 
      </span>
      <span
        className="splash__words two"
      >
        The answer is a 
      </span>
      <span
        className="splash__words three"
      >
        <i>
          memorable
        </i>
        , 
      </span>
      <span
        className="splash__words four"
      >
        <i>
          simplistic
        </i>
        , 
      </span>
      <span
        className="splash__words five"
      >
        <i>
          intuitive
        </i>
        , 
      </span>
      <span
        className="splash__words six"
      >
        and 
        <i>
          fast
        </i>
         
      </span>
      <span
        className="splash__words seven"
      >
        user experience. 
      </span>
      <span
        className="splash__words eight"
      >
        My goal is to create that perfect website for your business.
      </span>
    </p>
    <p
      className="splash__name"
    >
      Robby Bergers.
    </p>
    <button
      className="splash__to-site"
      onClick={[Function]}
    >
      Learn more →
    </button>
  </div>
</div>
`;
